integration/gethnetwork/geth_binaries.go

```diff
@@ -0,0 +1,41 @@
+package gethnetwork
+
+import (
+	"fmt"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"sync"
+)
+
+const (
+	shCmd             = "sh"
+	gethScriptPathRel = "./build_geth_binary.sh"
+	gethBinaryPathRel = "../.build/geth_bin/geth"
+	versionFlag       = "--version"
+
+	LatestVersion = "v1.10.17" // geths release version
+)
+
+var (
+	// prevents issues when calling from different packages/directories
+	_, b, _, _ = runtime.Caller(0)
+	basepath   = filepath.Dir(b)
+
+	creationLock sync.Mutex // makes sure there isn't two creations running at the same time
+)
+
+// EnsureBinariesExist makes sure geth binary exist, returns path where geth binaries exist
+// Runs the build_geth_binary.sh while handling source directory call
```

Comment:
I don't understand what "handling source directory call" means?

Reply:
Depending on where you execute the command there might be a relative source directory that might be unexpected.

Running code from `~/go/src/github.com/obscuronet/obscuro-playground` and from `~/go/src/github.com/obscuronet/obscuro-playground/go/obscurocommon` have different source directory calls.